src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4177,9 +4218,17 @@ private void LearnFromNonNullTest(BoundExpression expression, ref LocalState sta
         {
             if (expression is BoundValuePlaceholderBase placeholder)
             {
-                if (_resultForPlaceholdersOpt != null && _resultForPlaceholdersOpt.TryGetValue(placeholder, out var value))
+                if (_resultForPlaceholdersOpt != null)
```

Comment:
>_resultForPlaceholdersOpt != null [](http://example.com/codeflow?start=20&length=33)

Could you elaborate why the condition got split? Don't we want to assert if the dictionary is null?